### PR TITLE
Cleanup vk Initializers

### PR DESCRIPTION
**Pull request by McCallisterRomer**
Cleanup of functions in the `Initializers` file. No logic changes - just refactoring to rein in the file a bit.

I also created a `vk` subdirectory in graphics and moved (some) vulkan abstractions there. Ideally, all vulkan-related stuff will be housed here, eventually. This would make it easier to reason about where vulkan dependencies exist and identify leaks early. It also sets us up to refactor the whole caboodle to a separate library or repo, should we decide to go that route.

### CODE REVIEW

File: sdk/engine/source/graphics/Base.cpp [hunk 1]
```diff
@@ -1,9 +1,10 @@
 #include "Base.h"
-#include "graphics/Initializers.h"
+#include "graphics/vk/Initializers.h"
```
Comment (jaremieromer) [inline sdk/engine/source/graphics/Base.cpp]: can just be vk/Initializers.h and can go next to vk/Swapchain.h, or make vk/Swapchain.h graphics/vk/Swapchain.h

File: sdk/engine/source/graphics/Renderer.h [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #pragma once
 
 #include "graphics\GpuAssetsStorage.h"
-#include "graphics\Initializers.h"
+#include "graphics\vk/Initializers.h"
```
Comment (jaremieromer) [inline sdk/engine/source/graphics/Renderer.h]: will you set all path separators in includes in this file to be the same, in this case /

File: sdk/engine/source/graphics/resources/CubeMapManager.cpp [hunk 3]
```diff
@@ -43,11 +43,11 @@ namespace nc::graphics
     {
         assert(data.size() < m_maxCubeMapsCount && !data.empty());
 
-        m_cubeMapSampler = m_graphics->GetBasePtr()->CreateTextureSampler();
+        m_cubeMapSampler = graphics::CreateTextureSampler(m_graphics->GetBasePtr()->GetDevice(), vk::SamplerAddressMode::eRepeat);
```
Comment (jaremieromer) [inline sdk/engine/source/graphics/resources/CubeMapManager.cpp:46]: Nice having constructor like this

File: sdk/engine/source/graphics/techniques/PbrTechnique.cpp [hunk 4]
```diff
@@ -75,7 +75,7 @@ namespace nc::graphics
         pipelineCreateInfo.setPViewportState(&viewportState);
         auto rasterizer = CreateRasterizationCreateInfo(vk::PolygonMode::eFill, 1.0f);
         pipelineCreateInfo.setPRasterizationState(&rasterizer);
-        auto multisampling = CreateMulitsampleCreateInfo(m_base->GetMaxSamplesCount());
```
Comment (jaremieromer) [inline sdk/engine/source/graphics/techniques/PbrTechnique.cpp:78]: Good eye!

File: sdk/engine/source/graphics/vk/Swapchain.h [hunk 5]
```diff
@@ -5,7 +5,9 @@
 
 namespace nc::graphics
 {
-    class Base; class RenderTarget;
+    // How many frames can be rendered concurrently.
+    // Each frame requires its own pair of semaphores.
+    constexpr uint32_t MaxFramesInFlight = 2u;
```
Comment (jaremieromer) [inline sdk/engine/source/graphics/vk/Swapchain.h:10]: i LOVE that this is in swapchain

**Review by jaremieromer — APPROVED** [review]
This feels so good! Such a good step in the right direction

File: sdk/engine/source/graphics/Renderer.h (hunk 2 above)
Comment (McCallisterRomer) [reply sdk/engine/source/graphics/Renderer.h]: Yes, this looks like a global find and replace type situation.

File: sdk/engine/source/graphics/Base.cpp (hunk 1 above)
Comment (McCallisterRomer) [reply sdk/engine/source/graphics/Base.cpp]: fixed

**Pull request merged**